wrangler-core/src/main/java/io/cdap/wrangler/utils/ArithmeticOperations.java

```diff
@@ -24,380 +24,479 @@
-   */
-  public static Float minus(Float x, Float y) {
-    if (x == null || y == null) {
-      return null;
-    }
-    return x - y;
-  }
-
-  /**
-   * Arithmetic operation - Subtract a column from another column.
-   */
-  public static BigDecimal minus(BigDecimal x, BigDecimal y) {
-    if (x == null || y == null) {
-      return null;
-    }
-    return x.subtract(y).stripTrailingZeros();
-  }
-
-  /**
-   Arithmetic operation - Multiply a column with another column.
-   */
-  public static Integer multiply(Integer x, Integer y) {
-    if (x == null || y == null) {
-      return null;
-    }
-    return x * y;
-  }
-
-  /**
-   * Arithmetic operation - Multiply a column with another column.
-   */
-  public static Double multiply(Double x, Double y) {
-    if (x == null || y == null) {
-      return null;
-    }
-    return x * y;
-  }
-
-  /**
-   * Arithmetic operation - Multiply a column with another column.
-   */
-  public static Float multiply(Float x, Float y) {
-    if (x == null || y == null) {
-      return null;
-    }
-    return x * y;
-  }
+  // The Number types upon which arithmetic operations can be performed, in order of increasing generality
+  private enum NumberType {
+    SHORT(0),
+    INTEGER(1),
+    LONG(2),
+    FLOAT(3),
+    DOUBLE(4),
+    BIGDECIMAL(5);
 
-  /**
-   * Arithmetic operation - Multiply a column with another column.
-   */
-  public static BigDecimal multiply(BigDecimal x, BigDecimal y) {
-    if (x == null || y == null) {
-      return null;
-    }
-    return x.multiply(y);
-  }
+    public final int priority;
 
-  /**
-   * Arithmetic operation - Divide a column by another and return the quotient.
-   */
-  public static Integer divideq(Integer x, Integer y) {
-    if (x == null || y == null || y == 0) {
-      return null;
+    private NumberType(int priority) {
+      this.priority = priority;
     }
-    return x / y;
   }
 
   /**
-   * Arithmetic operation - Divide a column by another and return the quotient.
+   * Return the "largest" (e.g. most general) type of the given numbers.
+   * Used for operations applied to columns of multiple types, in order to determine the type of the output column.
+   * The order of increasing generality is Short < Integer < Long < Float < Double < BigDecimal.
+   * Returns null if any input is null.
+   * Throws an error if any input of type Byte is given (this is not currently supported for arithmetic ops).
    */
-  public static Double divideq(Double x, Double y) {
-    if (x == null || y == null || y == 0) {
-      return null;
+  private static NumberType getOutputType(Number... nums) throws DirectiveExecutionException {
+    NumberType outputType = NumberType.SHORT;
+    for (Number num : nums) {
+      if (num instanceof Byte) {
+        throw new DirectiveExecutionException("Input cannot be of type 'Byte'.");
+      } else if (num == null) {
+        return null;
+      } else if (num instanceof Integer && outputType.priority == 0) {
```

Comment:
I think this would be cleaner to read if the comparison was made using the NumberType priority.

You can implement a `getPriotity()` method in the Enum and use that to extract the priority value. Then, you can simply compare the current outputType priority with the priority for each of the expected output types.

For example: 

```
if (num instanceof Integer && NumberType.INTEGER.getPriority() > outputType.getPriority()) {
...
} else if (num instanceof Long && NumberType.LONG.getPriority() > outputType.getPriority()) 
```

This way, it's more understandable that you are trying to compare the type of NUM with the priority for the ourput types in the NumberType field.